beacon-chain/db/deposits.go

```diff
@@ -22,19 +23,22 @@ var (
 
 // InsertDeposit into the database. If deposit or block number are nil
 // then this method does nothing.
-func (db *BeaconDB) InsertDeposit(ctx context.Context, d *pb.Deposit, blockNum *big.Int) {
+func (db *BeaconDB) InsertDeposit(ctx context.Context, d *pb.Deposit, blockNum *big.Int, depositRoot [32]byte) {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.InsertDeposit")
 	defer span.End()
 	if d == nil || blockNum == nil {
 		log.WithFields(logrus.Fields{
-			"block":   blockNum,
-			"deposit": d,
+			"block":        blockNum,
+			"deposit":      d,
+			"deposit root": hex.EncodeToString(depositRoot[:]),
 		}).Debug("Ignoring nil deposit insertion")
 		return
 	}
 	db.depositsLock.Lock()
 	defer db.depositsLock.Unlock()
-	db.deposits = append(db.deposits, &depositContainer{deposit: d, block: blockNum})
+	// keep the slice sorted on insertion in order to avoid costly sorting on retrival.
+	heightIdx := sort.Search(len(db.deposits), func(i int) bool { return db.deposits[i].Block.Cmp(blockNum) >= 0 })
+	db.deposits = append(db.deposits[:heightIdx], append([]*DepositContainer{{deposit: d, Block: blockNum, DepositRoot: depositRoot}}, db.deposits[heightIdx:]...)...)
```

Comment:
this is a long line, maybe be break this up into two smaller ones. 
```suggestion
        newDeposits :=  append([]*DepositContainer{{deposit: d, Block: blockNum, DepositRoot: depositRoot}},db.deposits[heightIdx:]...)
	db.deposits = append(db.deposits[:heightIdx], newDeposits...)
```

Reply:
thanks